extractor/model.go

```diff
@@ -0,0 +1,21 @@
+package extractor
+
+// proteus:generate
+type Request struct {
+	RepositoryID   string
+	RootCommitHash []byte
+	Reference      string
+}
+
+// proteus:generate
+type RepositoryData struct {
+	RepositoryID   string
+	Files []File
+}
+
+type File struct {
+	Language string
+	Path     string
+	AST      string
+	Hash     []byte
```

Comment:
Is this file hash or commit hash? If it's commit hash, call it `Commit` or `CommitHash`?